tests/recipient_validation/test_phone_number.py

```diff
@@ -383,19 +385,19 @@ def test_format_uk_and_international_phone_numbers(phone_number, expected_format
         ("foo", "foo"),
         ("TeSt@ExAmPl3.com", "test@exampl3.com"),
         ("+4407900 900 123", "447900900123"),
-        ("+1 800 555 5555", "18005555555"),
+        ("+1 928-282-4541", "19282824541"),
     ],
 )
 def test_format_recipient(recipient, expected_formatted):
     assert format_recipient(recipient) == expected_formatted
 
 
-def test_try_format_recipient_doesnt_throw():
-    assert try_validate_and_format_phone_number("ALPHANUM3R1C") == "ALPHANUM3R1C"
+# def test_try_format_recipient_doesnt_throw():
+#     assert try_validate_and_format_phone_number("ALPHANUM3R1C") == "ALPHANUM3R1C"
```

Comment:
currently commented out, confirming we can delete as we don't have an equivalent method. Invalid characters would be caught at the instantiation of the object, not during a try_and_validate_and_format_method

Reply:
I think the point of the method is to not catch invalid characters.

Are we testing `format_recipient()` elsewhere? If we do, and we are sure nowhere else uses `try_validate_and_format_phone_number()`, then we can remove this test.

Reply:
It looks like we do use it in API repo: https://github.com/search?q=repo%3Aalphagov%2Fnotifications-api%20try_validate_and_format_phone_number&type=code

So looks like we need that method, or an equivalent.

Reply:
We'll do validating and formatting separately in API and Admin as it follows our new pattern. I don't think we should have a function that is equivalent to `try_validate_and_format_phone_number` our new validation flow should be:

```
number = PhoneNumber("ALPHANUM3R1C")
number.validate(<pass international and landline arguments as needed>) # optional, we may or may not want to check service permissions against the number
formatted_number = number.get_normalised_format()
```

Our new code forbids alphanumeric numbers:

```
    @staticmethod
    def _raise_if_phone_number_contains_invalid_characters(number: str) -> None:
        chars = set(number)
        if chars - {*ALL_WHITESPACE + "()-+" + "0123456789"}:
            raise InvalidPhoneError(code=InvalidPhoneError.Codes.UNKNOWN_CHARACTER)
```

We do however check that the human readable and normalised formats works correctly for valid numbers. Which I believe is the only thing we want to make sure works properly for the cases where human readable and normalised formates are needed in API and Admin

Reply:
> Our new code forbids alphanumeric numbers (and the above code snippet would fail on the first line with "ALPHANUM3R1C" as an argument

But we might be getting an alphanumeric sender for our inbound sms, right? According to [this commit](https://github.com/alphagov/notifications-api/commit/3daf039fbe58e8f517537ce49534e23cc10e6bc5#diff-846b9511257439f27f518ef400cd0c75e74f737087ad044b02b656e6798375d2R33) at least (it is a very old commit, so maybe not anymore? Not sure)

So does it mean API will have its own code that allows alphanumerics for that particular case?

Or is it a case where we thought we might need alphanumerics support back when implementing inbound sms in 2017, but it turned out we don't need them?

Reply:
I don't know. I know Leo has is explicitly raise an exception if we try now to it's not like it's an edge case where we didn't realise ... easy enough to add back in if needed though

Reply:
Decision record:

After a Slack discussion between Leo, Richard and me, we decided the following:

- we do need to support alphanumeric senders for inbound SMS
- we will move the validate and format functions from utils to admin and API, respectively (since one of the function has been used only by admin, and another was used just by API)
- these functions will parse the numbers by initialising them with PhoneNumber class, but if there is InvalidPhoneError, they will simply return the number, like the functions we are removing form utils do.